### PR TITLE
Kdesktop 1445 sync optimization sync db path

### PR DESCRIPTION
**Pull request by ChristopheLarchier**

### CODE REVIEW

File: src/server/appserver.cpp [hunk 1]
```diff
@@ -1909,6 +1909,7 @@ void AppServer::onRequestReceived(int id, RequestNum num, const QByteArray &para
         }
         case RequestNum::UTILITY_DISPLAY_CLIENT_REPORT: {
             sentry::pTraces::basic::AppStart().stop();
+            break;
```
Comment (luc-guyot-infomaniak) [inline src/server/appserver.cpp:1912]: It seems to be an unrelated fix. What actual issue was caused by the missing `break` statement?

File: src/libsyncengine/db/syncdb.cpp [hunk 2]
```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
...
-    if (!queryNext(SELECT_NODE_BY_NODEID_LITE_ID, found)) {
-        LOG_WARN(_logger, "Error getting query result: " << SELECT_NODE_BY_NODEID_LITE_ID << " - nodeId=" << dbNodeId);
-        return false;
-    }
-    if (!found) {
-        return true;
-    }
-
-    // Fill names' vector
+    // names' vector
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:987]:
```suggestion
    // Local and remote name vector.
```

File: src/libsyncengine/db/syncdb.cpp [hunk 3]
```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
...
+    ASSERT(queryBindValue(requestId, 1, dbNodeId));
 
-                ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMELOCAL, localName));
-                ASSERT(querySyncNameValue(SELECT_NODE_BY_NODEID_LITE_ID, SELECT_NODE_BY_NODEID_NAMEDRIVE, driveName));
-                names.push_back({localName, driveName});
-            }
+    for (;;) {
+        bool hasNext = false;
+        if (!queryNext(requestId, hasNext)) {
+            LOG_WARN(_logger, "Error getting query result: " << requestId.c_str());
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:998]:
```suggestion
            LOG_WARN(_logger, "Error getting query result: " << requestId);
```

File: src/libsyncengine/db/syncdb.cpp [hunk 4]
```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
...
+            return false;
         }
+        if (!hasNext) {
+            break;
+        }
+
+        SyncName nameLocal;
+        ASSERT(querySyncNameValue(requestId, 0, nameLocal));
+        SyncName nameDrive;
+        ASSERT(querySyncNameValue(requestId, 1, nameDrive));
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1008]:
```suggestion
        ASSERT(querySyncNameValue(requestId, 1, nameDrive));
```
```suggestion
        ASSERT(querySyncNameValue(requestId, 1, nameDrive)); // Name on the remote drive.
```

File: src/libsyncengine/db/syncdb.cpp [hunk 5]
```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
...
 
-    // Construct path from names' vector
-    localPath.clear();
-    remotePath.clear();
-    for (auto nameIt = names.rbegin(); nameIt != names.rend(); ++nameIt) {
-        localPath.append(nameIt->first);
-        remotePath.append(nameIt->second);
+    found = (names.size() > 0);
+
+    if (found) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1017]: We may omit it because the for-loop below is empty if `found` is `false`. Besides, clearing the paths at the very start would be a more predictable of the function call.

File: src/libsyncengine/db/syncdb.cpp [hunk 6]
```diff
@@ -1111,61 +984,44 @@ bool SyncDb::dbIds(std::unordered_set<DbNodeId> &ids, bool &found) {
...
-    ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
+    ASSERT(queryResetAndClearBindings(requestId));
 
-    // Construct path from names' vector
-    localPath.clear();
-    remotePath.clear();
-    for (auto nameIt = names.rbegin(); nameIt != names.rend(); ++nameIt) {
-        localPath.append(nameIt->first);
-        remotePath.append(nameIt->second);
+    found = (names.size() > 0);
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1015]:
```suggestion
    found = !names.empty();
```

File: src/libsyncengine/db/syncdb.cpp [hunk 7]
```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
...
-                                                         << (side == ReplicaSide::Local ? " - nodeIdLocal=" : " - nodeIdDrive=")
-                                                         << nodeId.c_str());
-        return false;
-    }
-    if (!found) {
-        return true;
-    }
-
-    // Fill names' vector
+    // names' vector
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1433]: We can remove this comment now as it only emphasizes the name of the variable.

File: src/libsyncengine/db/syncdb.cpp [hunk 8]
```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
...
     std::vector<SyncName> names;
 
-    bool parentNodeDbIdIsNull;
-    DbNodeId parentNodeDbId;
-    ASSERT(queryIsNullValue(id, SELECT_NODE_BY_REPLICAID_PARENTID, parentNodeDbIdIsNull));
-    if (!parentNodeDbIdIsNull) {
-        ASSERT(queryInt64Value(id, SELECT_NODE_BY_REPLICAID_PARENTID, parentNodeDbId));
+    std::string requestId =
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1436]:
```suggestion
    const std::string requestId =
```

File: src/libsyncengine/db/syncdb.cpp [hunk 9]
```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
...
-                id, side == ReplicaSide::Local ? SELECT_NODE_BY_REPLICAID_NAMELOCAL : SELECT_NODE_BY_REPLICAID_NAMEDRIVE, name));
-        names.push_back(name);
+    ASSERT(queryResetAndClearBindings(requestId));
+    ASSERT(queryBindValue(requestId, 1, nodeId));
 
-        ASSERT(queryResetAndClearBindings(id));
+    for (;;) {
+        bool hasNext = false;
+        if (!queryNext(requestId, hasNext)) {
+            LOG_WARN(_logger, "Error getting query result: " << requestId.c_str());
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1445]:
```suggestion
            LOG_WARN(_logger, "Error getting query result: " << requestId);
```

File: src/libsyncengine/db/syncdb.cpp [hunk 10]
```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
...
-                names.push_back(name);
-            }
+    ASSERT(queryResetAndClearBindings(requestId));
 
-            ASSERT(queryResetAndClearBindings(SELECT_NODE_BY_NODEID_LITE_ID));
-        }
-    } else {
-        ASSERT(queryResetAndClearBindings(id));
-    }
+    found = (names.size() > 0);
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1460]:
```suggestion
    found = !names.empty();
```

File: src/libsyncengine/db/syncdb.cpp [hunk 11]
```diff
@@ -1574,67 +1430,41 @@ bool SyncDb::parent(ReplicaSide side, const NodeId &nodeId, NodeId &parentNodeid
...
 
-    // Construct path from names' vector
-    path.clear();
-    for (auto nameIt = names.rbegin(); nameIt != names.rend(); ++nameIt) {
-        path.append(*nameIt);
+    if (found) {
```
Comment (luc-guyot-infomaniak) [inline src/libsyncengine/db/syncdb.cpp:1462]: Same remarks as above: I would not test the `found` value and I would alway clear `path`.

File: src/server/appserver.cpp (hunk 1 above)
Comment (ChristopheLarchier) [reply src/server/appserver.cpp:1912]: The next C/S request (SYNC_SETSUPPORTSVIRTUALFILES) is called, fortunately without parameters (so syncDbId=0).
=> Error in the log ""Error in setSupportsVirtualFiles for syncDbId=0"

File: src/libsyncengine/db/syncdb.cpp (hunk 2 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 3 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 4 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 6 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 7 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 8 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 9 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 10 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done

File: src/libsyncengine/db/syncdb.cpp (hunk 11 above)
Comment (ChristopheLarchier) [reply src/libsyncengine/db/syncdb.cpp]: Done
